odb_extractor_atoms.py

```diff
@@ -59,98 +223,297 @@ def select_obsclass(classes):
+                    if ii == 0:
+                        noffsets += 1
+                    else:
+                        if poffsets[ii] != poffsets[ii_prevobj] or qoffsets[ii] != qoffsets[ii_prevobj]:
+                            noffsets += 1
+                    if noffsets % 2 == 1:
+                        nextatom = True
+                        atomstr += 'offset pattern'
+                else:
+                    npattern -= 1
+                    if npattern < 0:
+                        nextatom = True
+                        atomstr += 'offset pattern'
+                        npattern = offset_lag - 1
+
+            ii_prevobj = ii
+
+        # New atom?
         if nextatom:
-            # Get class and qastate for previous atom
+            # Get class, qastate, guiding for previous atom
             if natom > 0:
-                print(qastates)
-                atoms[-1]['qastate'] = select_qastate(qastates)
-                print(classes)
+                atoms[-1]['qa_state'] = select_qastate(qastates)
+                if atoms[-1]['qa_state'] != 'NONE':
+                    atoms[-1]['observed'] = True
                 atoms[-1]['class'] = select_obsclass(classes)
-
-            # New atom
+                if verbose:
+                    print('QA states: ', qastates, file=fid)
+                    print('Classes: ', classes, file=fid)
+                    print('Guiding: ', guiding, file=fid)
+                atoms[-1]['guide_state'] = any(guiding)
+                atoms[-1]['wavelength'] = config['wavelength'][ii]
+                atoms[-1]['required_resources']['inst'] = config['inst']
+                atoms[-1]['required_resources']['filter'] = config['filter'][ii]
+                atoms[-1]['required_resources']['disperser'] = config['disperser'][ii]
+                atoms[-1]['required_resources']['fpu'] = config['fpu'][ii]
+
+                # Print basic atom info
+                print(" \t exec_time: {:7.2f}, prog_time: {:7.2f}, part_time: {:7.2f}, guide_state: {}".
+                      format(atoms[-1]['exec_time'], atoms[-1]['prog_time'], atoms[-1]['part_time'],
+                             atoms[-1]['guide_state']), file=fid)
+
+            # New atom entry
+            print(atomstr, file=fid)
             natom += 1
             atoms.append({'id': natom, 'exec_time': 0.0, 'prog_time': 0.0, 'part_time': 0.0,
-                          'class': 'NONE', 'qastate': 'NONE'})
+                          'class': 'NONE', 'observed': False, 'qa_state': 'NONE', 'guide_state': False,
+                          'wavelength': 0.0,
+                          'required_resources': {'inst': 'NONE', 'filter': 'NONE', 'disperser': 'NONE',
+                                                 'fpu': 'NONE'}})
             classes = []
+            guiding = []
+            qastates = []
+            if step['observe:observeType'].upper() in ['FLAT', 'ARC', 'DARK', 'BIAS'] \
```

Comment:
The list should probably be defined elsewhere, because as it stands, we recreate it every time we reach this statement, which is unnecessary.